internal/query/projection/user_metadata.go

```diff
@@ -49,6 +49,9 @@ func (*userMetadataProjection) Init() *old_handler.Check {
 		},
 			handler.NewPrimaryKey(UserMetadataColumnInstanceID, UserMetadataColumnUserID, UserMetadataColumnKey),
 			handler.WithIndex(handler.NewIndex("resource_owner", []string{UserGrantResourceOwner})),
+			handler.WithIndex(handler.NewIndex("metadata_key", []string{UserMetadataColumnKey})),
```

Comment:
Hey Marco, thanks for this work! Huge impact. 

Two things I'm thinking of when looking at these indexes: First, `metadata_key_and_value` covers a lone `metadata_key` lookup so this one isn't needed. 

Second, and more importantly, none of these b-tree indexes are sufficient for partial text searches (eg `... like %term%`) that are generated for `TEXT_FILTER_METHOD_CONTAINS` and the case insensitive version. We'll likely want a GIN trigram index to make this work efficiently for large deployments.

Reply:
Follow up - if we do end up using gin w/ `pg_trgm` we'll need to update the proto to enforce min string length of 3 for partial match searches otherwise Postgres will just fallback to a full scan.

Reply:
>  I was not sure whether an index with meta key + meta value was the same as having 1 index meta key and another on meta value. I guess I can remove the 2 indices on meta key and meta value?

Righto, for postgres multicolumn indexes, the first value(s) can be used alone. So an index "a,b,c" can be used as "a" and "a,b" alone, but not "b","c" or "b,c". So key,value counts for key, but not value alone